deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/resource/Schedules.java

```diff
@@ -0,0 +1,137 @@
+        scheduleDAO = context.getScheduleDAO();
+        environDAO = context.getEnvironDAO();
+        provisionHandler = new ProvisionHandler(context);
+    }
+
+    @GET
+    @Path("/{scheduleId : [a-zA-Z0-9\\-_]+}")
+    public ScheduleBean getSchedule(
+            @Context SecurityContext sc,
+            @PathParam("scheduleId") String scheduleId) throws Exception {
+
+        String operator = sc.getUserPrincipal().getName();
+
+        ScheduleBean scheduleBean = scheduleDAO.getById(scheduleId);
+        if (scheduleBean!=null) {
+            LOG.info(scheduleBean.toString());
+        }
+        return scheduleBean;
+    }
+    
+    @POST
+    @Path("/update/{envName : [a-zA-Z0-9\\-_]+}/{stageName : [a-zA-Z0-9\\-_]+}")
+    public void updateSchedule(
+            @Context SecurityContext sc,
+            @PathParam("envName") String envName,
+            @PathParam("stageName") String stageName,
+            @Valid ScheduleBean bean) throws Exception {
+        String operator = sc.getUserPrincipal().getName();
+        EnvironBean envBean = environDAO.getByStage(envName, stageName);
+        String scheduleId = envBean.getSchedule_id();
+        String cooldownTimes = bean.getCooldown_times();
+        String hostNumbers = bean.getHost_numbers();
+        Integer totalSessions = bean.getTotal_sessions();
+        if (totalSessions > 0) { // there is a schedule  
+            ScheduleBean scheduleBean = new ScheduleBean();
+            scheduleBean.setState_start_time(System.currentTimeMillis());
+            scheduleBean.setCooldown_times(cooldownTimes);
+            scheduleBean.setHost_numbers(hostNumbers);
+            scheduleBean.setTotal_sessions(totalSessions);
+            LOG.info(scheduleBean.toString());
+            if (scheduleId == null) {
+                scheduleId = CommonUtils.getBase64UUID();
+                envBean.setSchedule_id(scheduleId);
+                environDAO.update(envName, stageName, envBean);
+                scheduleBean.setId(scheduleId);
+                scheduleDAO.insert(scheduleBean);
+                LOG.info(String.format("Successfully inserted one env %s (%s)'s schedule by %s: %s", envName, stageName, operator, scheduleBean.toString()));
+            } else {
+                scheduleBean.setId(scheduleId);
+                scheduleDAO.update(scheduleBean, scheduleId);
+                LOG.info(String.format("Successfully updated one env %s (%s)'s schedule by %s: %s", envName, stageName, operator, scheduleBean.toString()));
+            }
+        } else if (scheduleId != null) { //there are no sessions, so delete the schedule
+            scheduleDAO.delete(scheduleId); 
+            environDAO.deleteSchedule(envName, stageName);
+            LOG.info(String.format("Successfully deleted env %s (%s)'s schedule by %s", envName, stageName, operator));
+        }
+    }
+
+    @POST
+    @Path("/override/{envName : [a-zA-Z0-9\\-_]+}/{stageName : [a-zA-Z0-9\\-_]+}")
```

Comment:
I would suggest url: /v1/env/stage/schedules/override?session_id=n and use HTTP PUT